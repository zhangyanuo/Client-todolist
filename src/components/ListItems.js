
import React, { useRef } from 'react'
import styled from 'styled-components'
import { Checkbox, Inline, Stack, TextInput } from "@construct-kit/core";

const CheckboxContainer = styled.div`
  border-bottom:1px solid #eee;
  display:flex;
  justify-content:space-between;
`

function ListItems(props) {
  const { list: { _id, name, isEdit, status }, delectItem, finished, changeToSave, changeToEdit } = props;
  const inputRef = useRef();
  const changeStatus = () => {
    finished(_id);
  }
  const deleted = () => {
    delectItem(_id);
  }

  const editDisplay = () => {
    changeToSave(_id);
  }

  const saveDisplay = () => {
    const txt = inputRef.current.value;
    changeToEdit(_id,txt);
  }

  return (
    <Stack inset="0.33rem" gap="small" key={_id}>
      <CheckboxContainer>
        {isEdit ?
          (<Checkbox label={name} checked={status} onChange={changeStatus} />) :
          (<TextInput ref={inputRef} label="" hideLabel defaultValue={name} ></TextInput>)
        }
        <Inline inset="0" gap="small" grow={false}>
          {isEdit ?
            (<button onClick={editDisplay} >edit</button>) :
            (<button onClick={saveDisplay}>save</button>)
          }
          <button onClick={deleted}>delete</button>
        </Inline>

      </CheckboxContainer>
    </Stack>
  )


}

export default ListItems;