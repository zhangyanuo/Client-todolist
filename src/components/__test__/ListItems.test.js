import ListItems from '../ListItems';
import React from 'react';
import { shallow } from 'enzyme';

test('render listItems component', () => {

    const list ={
        _id:'111',
        name:"aaa",
        isEdit:true,
        status:0
    }

    const wrapper = shallow(
        <ListItems list={list} key={list._id} />
    )
    expect(wrapper).toMatchSnapshot();
})