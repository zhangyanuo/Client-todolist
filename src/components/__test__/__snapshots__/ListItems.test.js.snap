// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render choose component 1`] = `ShallowWrapper {}`;

exports[`render listItems component 1`] = `ShallowWrapper {}`;
